envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -375,5 +399,6 @@ data class ClusterConfiguration(
 class RouteSpecification(
```

Comment:
Maybe instead of adding a clusterWeight Property, we should create WeightedRouteSpecification? It will simplify some code and we coud get rid of some checks if this map contains keys like main and secondary.